### PR TITLE
(/guides/basic-rbac): update code examples for Next 15; resolve a few minor issues; update copy

### PR DESCRIPTION
**Pull request by royanger**
> [!IMPORTANT]
> 🔎 Previews:
>
> - https://clerk.com/docs/pr/1669/guides/basic-rbac

[DOCS-9468](https://linear.app/clerk/issue/DOCS-9468/update-page-protection-section-of-rbac-guide)

### Explanation:

Updated the code examples to be consistent and work with Next 15 and @clerk/nextjs v6

-

### CODE REVIEW

**Comment by OwaisSiddiqui** [comment]
I think there is an error in the commits after working on my code:

I think it should change from `await clerk.users` to `await client.users` under Admin tools section (since it's now `const client = await clerkClient()`).

**Comment by royanger** [comment]
> I think there is an error in the commits after working on my code:
> 
> I think it should change from `await clerk.users` to `await client.users` under Admin tools section (since it's now `const client = await clerkClient()`).

Good catch. I fixed that.

File: docs/guides/basic-rbac.mdx [hunk 1]
```diff
@@ -239,12 +228,11 @@ This guide assumes that you're using Next.js App Router, but the concepts can be
...
 
-  If one or more users are found, the component will display a list of users, showing their first and last names, primary email address, and current role. Each user will have `Make Admin` and `Make Moderator` buttons, which include hidden inputs for the user ID and role. These buttons will use the `setRole()` server action to update the user's role.
+  With the server action and the search form set up, it's time to refactor the `app/admin/page.tsx`.
 
-  1. Navigate to your `app/admin/page.tsx` file.
-  1. Replace the code with the following code.
+  1. Replace the code in your `app/admin/page.tsx` file with the following code. It checks whether a search parameter has been appended to the URL by the search form. If a search parameter is present, it queries for users matching the entered term. If one or more users are found, the component displays a list of users, showing their first and last names, primary email address, and current role. Each user has `Make Admin` and `Make Moderator` buttons, which include hidden inputs for the user ID and role. These buttons use the `setRole()` server action to update the user's role.
 
   ```tsx {{ filename: 'app/admin/page.tsx' }}
```
Comment (alexisintech) [inline docs/guides/basic-rbac.mdx:237]: This example is giving a type error! I think the actions need to return `void` or `Promise<void>`, but the actions are returning an object with a message property.

Screenshot 2024-11-13 at 17 30 27

Comment (royanger) [reply docs/guides/basic-rbac.mdx:237]: Can you share your tsconfig and I can take a look into this.

Comment (alexisintech) [reply docs/guides/basic-rbac.mdx:237]:
```
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "plugins": [
      {
        "name": "next"
      }
    ],
    "paths": {
      "@/*": ["./*"]
    }
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}

```

Comment (royanger) [reply docs/guides/basic-rbac.mdx:237]: I just pushed a commit that resolves this on my end. Let me know if you're still seeing errors.

Comment (alexisintech) [reply docs/guides/basic-rbac.mdx:237]: yeah it still gives me those type errors, but if you are saying you don't have the type errors then it should be fine!

**Review by alexisintech — APPROVED** [review]

File: docs/guides/basic-rbac.mdx (hunk 1 above)
Comment (royanger) [reply docs/guides/basic-rbac.mdx:237]: Can you share your repo? The errors should be gone. There is either something in your repo (and maybe still in this code that I didn't update) or there is something that is suppressing the errors in my editor.

Comment (alexisintech) [reply docs/guides/basic-rbac.mdx:243]: It is this repo: [aa-clerk-next-app](https://github.com/alexisintech/aa-clerk-next-app)
but I made a repo based off of our updated demo repo, and it works with no errors https://github.com/alexisintech/aa-clerk-nextjs-demo2

Comment (alexisintech) [reply docs/guides/basic-rbac.mdx:243]: so I wouldn't worry about it!

**Comment by alexisintech** [comment]
I think this is good to go whenever! @royanger

**Pull request merged**